ce_main_app/ikbd/ikbd.h

```diff
@@ -99,10 +105,20 @@ class Ikbd
     int     joy1st;
     int     joy2nd;
 
+    bool    firstJoyIs0;                    // if true, joystick keyboard mapping and usb/atari joysticks are swapped
     bool    mouseWheelAsArrowsUpDown;       // if true, mouse wheel up / down will be translated to arrow up / down
     bool    keybJoy0;                       // if true, specific keys will act as joy 0
     bool    keybJoy1;                       // if true, specific keys will act as joy 1
 
+    // this is basically all keys which are used for the hotswap, perhaps
+    // it should be made a bit more flexible, for different key combinations
+    int     leftShiftsPressed;  // how many LEFT SHIFTs are pressed (1x for PC, 1x for Atari)
+    int     rightShiftsPressed; // how many RIGHT SHIFTs are pressed (1x for PC, 1x for Atari)
+    int     ctrlsPressed;       // how many C[ON]TR[O]Ls are pressed (2x for PC, 1x for Atari)
+    int     f11sPressed;        // HELP on Atari, F11 on PC
+    int     f12sPressed;        // UNDO on Atari, F12 on PC
+    bool    waitingForHotkeyRelease;
```

Comment:
Because I was lazy (started only with ctrlPressed and shiftPressed), then the situation got out of hand and then you came with your elegant solution and I felt ashamed ;) (and was afraid to change it not to break anything -- remember, you're collecting only _USB_ keystrokes while I collect from both keyboards).